src/utils/integration_test.rs

```diff
@@ -0,0 +1,157 @@
+#![cfg(test)]
+
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use axum::Router;
+use tracing::info;
+
+use crate::bus::metrics::BusMetrics;
+use crate::bus::SharedMessageBus;
+use crate::consensus::Consensus;
+use crate::data_availability::DataAvailability;
+use crate::indexer::Indexer;
+use crate::mempool::Mempool;
+use crate::model::{CommonRunContext, NodeRunContext, SharedRunContext};
+use crate::p2p::P2P;
+use crate::single_node_consensus::SingleNodeConsensus;
+use crate::utils::conf::Conf;
+use crate::utils::crypto::BlstCrypto;
+use crate::utils::logger::LogMe;
+use crate::utils::modules::ModulesHandler;
+
+pub struct NodeIntegrationCtxBuilder {
+    pub conf: Conf,
+    pub bus: SharedMessageBus,
+    pub crypto: BlstCrypto,
+}
+
+impl NodeIntegrationCtxBuilder {
+    pub async fn new() -> Self {
+        let tmpdir = tempfile::tempdir().unwrap().into_path();
+        let bus = SharedMessageBus::new(BusMetrics::global("default".to_string()));
+        let crypto = BlstCrypto::new("test".to_owned());
+        let conf =
+            Conf::new(None, tmpdir.to_str().map(|s| s.to_owned()), Some(false)).expect("conf ok");
+
+        Self { conf, bus, crypto }
+    }
+
+    pub async fn build(self) -> Result<NodeIntegrationCtx> {
+        let conf = Arc::new(self.conf);
+        let mut node_modules = NodeIntegrationCtx::start_node(
+            conf.clone(),
+            self.bus.new_handle(),
+            self.crypto.clone(),
+        )
+        .await?;
+
+        // Just a little bit of tomfoolery
+        let nm = unsafe { NodeIntegrationCtx::lifetime_transmute(&mut node_modules) };
+        let starter = nm.start_modules();
+        let node_task = Some(tokio::spawn(starter));
+
+        // Wait a little bit for the node to start
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
this can lead into flakky test, is it possible to have a more robust way ? In the same idea as the wait_height in e2e ?